controller/RestEligibilities.php

```diff
@@ -0,0 +1,144 @@
+     *         name="delivery",
+     *         in="query",
+     *         description="Delivery Uri (Url encoded)",
+     *         required=true,
+     *         @OA\Schema(
+     *             type="string",
+     *         )
+     *     ),
+     *     @OA\Parameter(
+     *         name="testCenter",
+     *         in="query",
+     *         description="Test center Uri (Url encoded)",
+     *         required=true,
+     *         @OA\Schema(
+     *             type="string",
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response="200",
+     *         description="Eligibility data",
+     *         @OA\MediaType(
+     *             mediaType="application/json",
+     *             @OA\Schema(
+     *                 ref="#/components/schemas/Eligibility",
+     *                 example= {
+     *                     "delivery": "http://sample/first.rdf#i1536680377163170",
+     *                     "testCenter": "http://sample/first.rdf#i1536680377163171",
+     *                     "testTakers": {
+     *                         "http://sample/first.rdf#i1536680377163172",
+     *                         "http://sample/first.rdf#i1536680377163173"
+     *                     }
+     *                 }
+     *             )
+     *         ),
+     *     ),
+     *     @OA\Response(
+     *         response=404,
+     *         description="Eligibility not found",
+     *         @OA\MediaType(
+     *             mediaType="application/json",
+     *             @OA\Schema(
+     *                 example={
+     *                     "success": false,
+     *                     "errorCode": 404,
+     *                     "errorMsg": "Eligibility not found for provided search parameters.",
+     *                     "version": "3.3.0-sprint85"
+     *                 }
+     *             )
+     *         ),
+     *     ),
+     * )
+     */
+    public function get()
+    {
+        try {
+            $delivery = $this->getDeliveryFromRequest();
+            $testCenter = $this->getTCFromRequest();
+
+            /** @var EligibilityService $eligibilityService */
+            $eligibilityService = $this->getServiceLocator()->get(EligibilityService::SERVICE_ID);
+            $eligibility = $eligibilityService->getEligibility($testCenter, $delivery);
+
+            if ($eligibility === null) {
+                $result = [];
+            } else {
+                $result = $this->propagate(new Eligibility($eligibility->getUri()));
+            }
+
+            $this->returnJson([
+                'success' => true,
+                'data' => $result
+            ]);
+        } catch (\Exception $e) {
+            return $this->returnFailure($e);
+        }
+    }
+
+    /**
+     * Get delivery resource from request parameters
+     *
+     * @return \core_kernel_classes_Resource
+     * @throws common_exception_RestApi
+     */
+    private function getDeliveryFromRequest()
```

Comment:
This function is already exist in `RestEligibility` class. Would be better to move it into the abstract class. What do you think?

Reply:
This abstract class is common for few other controllers as well, which do not need delivery. Therefore I wouldn't move it. But it's just my personal opinion